crates/ty_python_semantic/src/types/class_base.rs

```diff
@@ -170,7 +171,11 @@ impl<'db> ClassBase<'db> {
                 | KnownInstanceType::TypeVar(_)
                 | KnownInstanceType::Deprecated(_)
                 | KnownInstanceType::Field(_)
-                | KnownInstanceType::ConstraintSet(_) => None,
+                | KnownInstanceType::ConstraintSet(_)
+                // A class inheriting from a newtype would make intuitive sense, but newtype
+                // wrappers are just identity functions at runtime, so this sort of inheritance
+                // doesn't work and isn't allowed.
+                | KnownInstanceType::NewType(_) => None,
```

Comment:
the runtime actually has a special-cased error message specifically for this, which we could consider emulating:

```pycon
>>> from typing import NewType
>>> X = NewType("X", int)
>>> class Foo(X): ...
... 
Traceback (most recent call last):
  File "<python-input-2>", line 1, in <module>
    class Foo(X): ...
  File "/Users/alexw/.pyenv/versions/3.13.1/lib/python3.13/typing.py", line 3428, in __init_subclass__
    raise TypeError(
    ...<2 lines>...
    )
TypeError: Cannot subclass an instance of NewType. Perhaps you were looking for: `Foo = NewType('Foo', X)`
```

But that obviously doesn't need to be done in this PR!